### PR TITLE
chore(search-bar): cleaned up props

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
Contributes to #4602

This PR cleans up the SearchBar props that were mentioned [here](https://github.com/carbon-design-system/ibm-products/issues/4239#issuecomment-1965282111). Groups up MultiSelect props for adopter ease of use. 

#### What did you change?
`placeHolderText` -> `placeholderText` (changed upper to lower case `h`)

#### How did you test and verify your work?
Changes work in my local Storybook instance

### CODE REVIEW

File: packages/ibm-products/src/components/SearchBar/SearchBar.js [hunk 1]
```diff
@@ -116,6 +116,15 @@ export let SearchBar = React.forwardRef(
       onChange(eventObject);
     };
 
+    const multiSelectProps = {
+      label: scopesTypeLabel,
+      initialSelectedItems: selectedScopes,
+      items: scopes,
+      itemToString: scopeToString,
+      translateWithId,
+      ...rest
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/SearchBar/SearchBar.js:125]: i don't think the rest here is necessary

**Review by davidmenendez — CHANGES_REQUESTED** [review]
aside from the lint error just one small comment

**Comment by IgnacioBecerra** [comment]
@davidmenendez Sorry for the wait! I was busy fixing a problem we've had in Dotcom for a while. Let me know if there's anything else so we can get this merged!

**Comment by matthewgallo** [comment]
@IgnacioBecerra there are some failing `SearchBar` tests now, see [here](https://github.com/carbon-design-system/ibm-products/actions/runs/8886189184/job/24399179313?pr=4923#step:6:242).

**Review by davidmenendez — APPROVED** [review]

**Pull request merged**